asf_levies_model/consumers.py

```diff
@@ -0,0 +1,460 @@
+        unit_converter : float, optional
+            Value to divide electricity and gas consumption values from dataframe by to convert to MWh, by default 1.
+
+        Returns
+        -------
+        Consumer
+            Consumer instance.
+        """
+
+        def safe_get(col_name, default=None):
+            """Helper function to retrieve column value if column name is provided."""
+            return df.loc[row, col_name] if col_name else default
+
+        name = safe_get(name_col)
+        archetype = safe_get(archetype_col)
+        net_annual_income = safe_get(net_annual_income_col)
+        main_heating_fuel = safe_get(main_heating_fuel_col)
+        gas_consumption = safe_get(gas_consumption_col, 0) / unit_converter
+        electricity_consumption = (
+            safe_get(electricity_consumption_col, 0) / unit_converter
+        )
+        net_income_decile = safe_get(net_income_decile_col, None)
+        unmetered_fuel_spend = safe_get(unmetered_fuel_spend_col, 0)
+        scheme_eligible = safe_get(scheme_eligible_col, False)
+        size = safe_get(size_col, 0)
+
+        return cls(
+            name=name,
+            archetype=archetype,
+            net_annual_income=net_annual_income,
+            net_income_decile=net_income_decile,
+            main_heating_fuel=main_heating_fuel,
+            gas_consumption=gas_consumption,
+            electricity_consumption=electricity_consumption,
+            unmetered_fuel_spend=unmetered_fuel_spend,
+            scheme_eligible=scheme_eligible,
+            size=size,
+            gas_tariff=gas_tariff,
+            electricity_tariff=electricity_tariff,
+        )
+
+    def __repr__(self):
+        """Printable representation of Consumer instance."""
+        attributes = [
+            (key, value)
+            for key, value in self.__dict__.items()
+            if not key.startswith("_")
+        ]
+        properties = [
+            (key, getattr(self, key))
+            for key, value in self.__class__.__dict__.items()
+            if isinstance(value, property)
+        ]
+        all_items = attributes + properties
+        repr_string = f"{self.__class__.__name__}("
+        repr_string += ", ".join([f"{key}={repr(value)}" for key, value in all_items])
+        repr_string += ")"
+        return repr_string
+
+    def boiler_to_hp_savings(
+        self,
+        boiler_efficiency: float = 0.85,
+        hp_efficiency: float = 3.0,
+        heating_pct: float = 0.97,
+    ) -> float:
+        """Returns estimated savings, in £, if household is assumed to switch from a gas boiler for heat to an electric heat pump.
+
+        Parameters
+        ----------
+        boiler_efficiency : float, optional
+            Assumed efficiency of gas boiler, by default 0.85
+        hp_efficiency : float, optional
+            Assumed efficiency of heat pump, by default 3.0
+        heating_pct : float, optional
```

Comment:
Assuming this is heating and hot water, as heating alone is ~77% of gas consumption.